lib/conjur/feature_flags/features.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Conjur
+  module FeatureFlags
+    # TODO: Add ability to turn all flags on by default (to insure tests can) be run against flagged features
+    class Features
+      # Instantiates a new Conjur::FeatureFlags::Features object
+      #
+      # @param logger [Logger] the logger used for output
+      # @param environment_settings [Hash] the environment variable has (typically ENV)
+      # @param feature_flags [Array<Symbol>] the list of valid feature flags
+      #
+      # @return [Conjur::FeatureFlags::Features]
+      def initialize(logger:, environment_settings:, feature_flags: [])
+        @logger = logger
+        @enabled_features = merge_with_environment_variable_flags(
+          features: feature_flags,
+          environment: environment_settings
+        )
+      end
+
+      # Check to see if a feature flag has been enabled
+      #
+      # @param feature_name [Symbol] the feature flag we want to check
+      #
+      # @return [Boolean]
+      def enabled?(feature_name)
+        @logger.debug("Conjur::FeatureFlags::Features#enabled? - feature flag name: '#{feature_name}'")
+        result = @enabled_features.key?(feature_name.downcase.to_sym) &&
+          @enabled_features[feature_name.downcase.to_sym]
+        @logger.debug("Conjur::FeatureFlags::Features#enabled? result: '#{result}'")
+        result
+      end
+
+      private
+
+      # Builds a hash of the feature flags whether they should be toggled on or
+      # off. Environment variable flags are set with the following format:
+      #   CONJUR_FEATURE_<feature_name>_ENABLED
+      #
+      # @param features [Array<Symbol>] list of feature flags
+      # @param environment [Hash] the environment variables and their values
+      #
+      # @return [Hash<Symbol>:<Boolean>] hash of feature names and their enabled state (true/false)
+      def merge_with_environment_variable_flags(features:, environment:)
+        {}.tap do |enabled_features|
+          features.each do |feature|
+            enabled_features[feature] = false
+            next unless environment["CONJUR_FEATURE_#{feature.upcase}_ENABLED"].to_s.downcase == 'true'
+
+            enabled_features[feature] = true
+            @logger.debug("Conjur::FeatureFlags::Features#merge_with_environment_variable_flags:")
+            @logger.debug("  feature '#{feature}' enabled via CONJUR_FEATURE_#{feature.upcase}_ENABLED")
+          end
```

Comment:
Purely functional alternative:
```ruby
def merge_with_environment_variable_flags(features:, env:)
  features.map do |f|
    enabled = env["CONJUR_FEATURE_#{f.upcase}_ENABLED"].downcase == 'true'
    log_reason_enabled(f) if enabled
    [f, enabled]
  end.to_h
end

def log_reason_enabled(feature)
  @logger.debug(
    "Feature '#{feature}' enabled via CONJUR_FEATURE_#{feature.upcase}_ENABLED"
  )
end
```

Reply:
@jonahx, thanks for this suggestion. I implemented it with the addition of stringifying `env["CONJUR_FEATURE_#{f.upcase}_ENABLED"]` to ensure we have a string if the key is not present in the hash.